tests/test_postgresql.py

```diff
@@ -19,6 +22,16 @@ def test_postgresql_proc(postgresql_proc: PostgreSQLExecutor) -> None:
     assert postgresql_proc.running() is True
 
 
+def test_postgresql_proc__bad_tmp(postgresql_proc: PostgreSQLExecutor) -> None:
+    r"""Test a \ character in the pytest temporary path."""
+
+    def bad_path(tmp_path_factory: TempPathFactory, subdir: str) -> Path:
+        return tmp_path_factory.mktemp(rf"bad\path/{subdir}")
+
+    with mock.patch("pytest_postgresql.factories.process._tmp_dir", side_effect=bad_path):
+        assert postgresql_proc.running() is True
```

Comment:
Hmmmm... looking at the test here. Executor at this stage is already started, it is running when you run first test with postgresql_proc.
It would be better I think to add test in test_executors, and instantiate the PostgreSQL executor there directly.
making the function `_tmp_dir` obsolete.

Reply:
good point. Thanks a lot! Done.

The new test has quite a bit of duplicate with the `test_executor_init_with_password` test.
It may be reasonable to refactor some more.

Reply:
If you want  I can open a new pull request [to refactor](https://github.com/bhelgs/pytest-postgresql/commit/c1929af82c56fe81455568791b625e00b5033d3f#diff-e8b376bb67834a23d818191fe7a2accaf5a9e36fb4f9ee8d925012406d207465L72).
This change damages test encapsulation, but at least I think it might resolve the `skipif` issue.

Reply:
I was just going to say that I did deduplicated these tests when fixing #343, but I was unable to duplicate the issue in tests, and just reverted any changes I made to tests 🤔 

Refactor looks good to me, note that port_for already contains PortType definition you can import and use.

Reply:
Sorry, I was on vacation and then got a bit distracted.

I'll treat the refactor as a separate PR.
Also, I was able to reproduce and add a test for what you fixed with PR #965.


What I did was add (`nl_NO.UTF-8`) and confirm it fails without the fix you made in PR #965:
<img width="635" alt="image" src="https://github.com/user-attachments/assets/bf9b0f86-f7e7-48dc-a3b1-705cce9a7a49">
`
(Not all languages encounter the issue that you fixed)

Reply:
done.